interface/src/app/analyze/analysis/analysisDetail.tpl.html

```diff
@@ -1,71 +1,75 @@
 <!--
   TODO each sample should be built using a reusable Angular directive
 -->
-<h1>Samples
-  <small>(drag <i class="fa fa-bars" aria-hidden="true"></i> to reorder,
-    click <i class="fa fa-times" aria-hidden="true"></i> to delete,
-    choose <input type="radio" disabled> to set group)</small>
-</h1>
-<status-bar msg-text="analysis.status"></status-bar>
-<!-- TODO need a tool to select and order annotation columns -->
-<table class="table table-hover">
-  <thead>
-    <tr>
-      <th> </th>
-      <th>
-        <span class="row-control">
-          <i class="fa fa-bars" aria-hidden="true"></i>
-        </span>
-        ML Data Source
-      </th>
-      <th>id</th>
-      <th>Sample Name</th>
-      <th ng-repeat="at in analysis.exampleCols">{{
-        at.typename
-      }}</th>
-    </tr>
-  </thead>
-  <tbody ng-model="sb.heatmapData.samples" as-sortable="sortableOptions">
-    <tr ng-repeat="id in sb.heatmapData.samples"
-        ng-mouseover="showControls = true" ng-mouseleave="showControls = false"
-        ng-class="'sample-' + sb.sampleToGroup[id]" as-sortable-item>
-      <td class="row-control">
-        <input type="radio" ng-model="sb.sampleToGroup[id]"
-            name="{{id}}-group" value="base-group"
-            class="sample-base-group" aria-label="group a">
-        <input type="radio" ng-model="sb.sampleToGroup[id]"
-            name="{{id}}-group" value="comp-group"
-            class="sample-comp-group" aria-label="group b">
-        <input type="radio" ng-model="sb.sampleToGroup[id]"
-            name="{{id}}-group" value="other"
-            class="sample-other" aria-label="other">
-        &nbsp;
-        <button type="button" class="btn row-control"
-            ng-click="sb.removeSample(id)" aria-label="Delete">
-          <i class="fa fa-times" aria-hidden="true"></i>
-        </button>
-      </td>
-      <td as-sortable-item-handle>
-        <span class="row-control">
-          <i class="fa fa-bars" aria-hidden="true"></i>
-        </span>
-        {{sb.sampleData[id].ml_data_source}}
-      </td>
-      <td>{{id}}</td>
-      <td>{{sb.sampleData[id].name}}</td>
-      <td ng-repeat="at in analysis.exampleCols">{{
-        sb.sampleData[id].annotations[at.typename]
-      }}</td>
-    </tr>
-  </tbody>
-</table>
-<p>To proceed with further analysis, please select a machine learning model.</p>
-<label>Machine learning model:</label>
-<ml-model-selector
-  selected-ml-model="sb.selectedMlModel"
```

Comment:
I expected that taking the `SampleBin` (`sb`, here) out of the loop on the `selectedMlModel` would cause serious trouble, but I'm not seeing it break on the demo server.

Just the same, I'm going to file an issue to update the `SampleBin` to defer to the `MlModelTracker`, since that is where we are now maintaining the state of that selection.